pkg/controller/deployment/deployment_controller.go

```diff
@@ -335,8 +357,16 @@ func (dc *DeploymentController) deletePod(obj interface{}) {
 		}
 	}
 	if d := dc.getDeploymentForPod(pod); d != nil && d.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
-		podList, err := dc.listPods(d)
-		if err == nil && len(podList.Items) == 0 {
+		// Sync if this Deployment now has no more Pods.
+		rsList, err := dc.getReplicaSetsForDeployment(d)
```

Comment:
Yes, respecting owner references is the reason for this PR and I am all for that ;)